default/AI/ColonisationAI.py

```diff
@@ -773,7 +763,33 @@ def evaluate_planet(planet_id, mission_type, spec_name, empire, detail=None):
     for bld_type in set(planet_build_names).intersection(AIDependencies.building_supply):
         planet_supply += sum(
             [AIDependencies.building_supply.get(bld_type, {}).get(int(psize), 0) for psize in [-1, planet.size]])
-    planet_supply += foAI.foAIstate.misc.get('supply_tech', 0)
+    for _special in set(planet.specials).intersection(AIDependencies.SUPPLY_MOD_SPECIALS):
+        planet_supply += sum(
+            [AIDependencies.SUPPLY_MOD_SPECIALS.get(_special, {}).get(int(psize), 0) for psize in [-1, planet.size]])
```

Comment:
`AIDependencies.SUPPLY_MOD_SPECIALS.get(_special, {})` `get` will always return value you can use `[]` in that case.

You pass `list` to `sum`, but you can use `generator expression` instead of list. And use syntax sugar to remove extra brackets:

`sum([x**2 for x in range(10)])` - > `sum((x**2 for x in range(10)))` -> `sum(x**2 for x in range(10))`   https://www.python.org/dev/peps/pep-0289